Sources/SwiftDocCUtilities/Action/Actions/Convert/ConvertAction.swift

```diff
@@ -357,6 +357,11 @@ public struct ConvertAction: Action, RecreatingContext {
     /// Converts each eligible file from the source documentation bundle,
     /// saves the results in the given output alongside the template files.
     mutating public func perform(logHandle: LogHandle) throws -> ActionResult {
+        defer {
+            diagnosticEngine.finalize()
+        }
+        
+        var convertEncounteredProblems: [Problem] = []
```

Comment:
How do you feel about naming this something like `postConversionProblems` and maybe adding a comment that explains why it's a separate variable? 
```suggestion
        // The converter has already emitted its problems to the diagnostic engine. Track additional problems separately to avoid repeating the converter's problems.
        var postConversionProblems: [Problem] = []
```